v compiler doesn't recognize interface type when passing to array method

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.7 18eee34.4ee948f
OS: windows, Microsoft Windows 11 Pro v22631 64-bit
Processor: 24 cpus, 64bit, little endian, 

getwd: D:\vbug
vexe: C:\v\v.exe
vexe mtime: 2024-09-24 17:12:18

vroot: OK, value: C:\v
VMODULES: OK, value: C:\Users\info\.vmodules
VTMP: OK, value: C:\Users\info\AppData\Local\Temp\v_0

Git version: git version 2.42.0.windows.2
Git vroot status: weekly.2024.39-5-g4ee948fe-dirty
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,

operable program or batch file.


thirdparty/tcc status: thirdparty-windows-amd64 b425ac82

```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg src\main.v`
```v
module main

interface IGameObject {
	mut:
	name string
}

struct GameObject implements IGameObject {
mut:
	name string
}

struct Game {
mut:
	objects        []IGameObject
}

fn (mut game Game) gc() {
	for obj in game.objects {
		game.objects.delete(obj)
	}
	game.objects.clear()
}

fn main() {}
```

**What did you expect to see?**

no c compiler error


**What did you see instead?**
```
================== C compilation error (from tcc): ==============
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01J8JMB0XJMWQ088W3RH9WNT80.tmp.c:7068: warning: implicit declaration of function 'tcc_backtrace'
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01J8JMB0XJMWQ088W3RH9WNT80.tmp.c:13536: error: cannot convert 'struct main__IGameObject' to 'int'
... (the original output was 3 lines long, and was truncated to 2 lines)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -3359,6 +3359,14 @@ fn (mut c Checker) array_builtin_method_call(mut node ast.CallExpr, left_type as
 		if method := c.table.find_method(unwrapped_left_sym, method_name) {
 			node.receiver_type = method.receiver_type
 		}
+		if node.args.len != 1 {
+			c.error('`.delete()` expected 1 argument, but got ${node.args.len}', node.pos)
+		} else {
+			arg_typ := c.expr(mut node.args[0].expr)
+			c.check_expected_call_arg(arg_typ, ast.int_type, node.language, node.args[0]) or {
+				c.error('${err.msg()} in argument 1 to `.delete()`', node.args[0].pos)
+			}
+		}
 		node.return_type = ast.void_type
 	}
 	return node.return_type

--- a/vlib/v/checker/tests/array_delete_arg_mismatch_err.out
+++ b/vlib/v/checker/tests/array_delete_arg_mismatch_err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/array_delete_arg_mismatch_err.vv:18:23: error: cannot use `IGameObject` as `int` in argument 1 to `.delete()`
+   16 | fn (mut game Game) gc() {
+   17 |     for obj in game.objects {
+   18 |         game.objects.delete(obj)
+      |                             ~~~
+   19 |     }
+   20 |     game.objects.clear()

--- a/vlib/v/checker/tests/array_delete_arg_mismatch_err.vv
+++ b/vlib/v/checker/tests/array_delete_arg_mismatch_err.vv
@@ -1,0 +1,23 @@
+interface IGameObject {
+mut:
+	name string
+}
+
+struct GameObject implements IGameObject {
+mut:
+	name string
+}
+
+struct Game {
+mut:
+	objects []IGameObject
+}
+
+fn (mut game Game) gc() {
+	for obj in game.objects {
+		game.objects.delete(obj)
+	}
+	game.objects.clear()
+}
+
+fn main() {}

